src/main/java/io/stargate/sgv2/jsonapi/api/model/command/builders/SortClauseBuilder.java

```diff
@@ -64,178 +57,67 @@ public SortClause build(JsonNode node) {
-              "Sort ordering value can only be `1` for ascending or `-1` for descending (not `%s`)",
-              inner.getValue());
-        }
-
-        boolean ascending = inner.getValue().intValue() == 1;
-        SortExpression exp = SortExpression.sort(validatedPath, ascending);
-        sortExpressions.add(exp);
-      }
+      sortExpressions.add(
+          buildAndValidateExpression(validateSortClausePath(path), inner.getValue(), totalFields));
     }
     return new SortClause(sortExpressions);
   }
 
-  private String validateSortClausePath(String path) {
-    if (!NamingRules.FIELD.apply(path)) {
-      if (path.isEmpty()) {
-        throw ErrorCodeV1.INVALID_SORT_CLAUSE_PATH.toApiException(
-            "path must be represented as a non-empty string");
+  protected abstract SortExpression buildAndValidateExpression(
+      String validatedPath, JsonNode innerValue, int totalFields);
+
+  protected SortExpression defaultBuildAndValidateExpression(
+      String validatedPath, JsonNode innerValue) {
+    if (!innerValue.isInt()) {
+      // Special checking for String and ArrayNode to give less confusing error messages
+      if (innerValue.isArray()) {
+        throw ErrorCodeV1.INVALID_SORT_CLAUSE_VALUE.toApiException(
+            "Sort ordering value can be Array only for Vector search");
       }
-      // But allow "well-known" fields
-      switch (path) {
-        case DocumentConstants.Fields.LEXICAL_CONTENT_FIELD,
-            DocumentConstants.Fields.VECTOR_EMBEDDING_FIELD,
-            DocumentConstants.Fields.VECTOR_EMBEDDING_TEXT_FIELD -> {}
-        default ->
-            throw ErrorCodeV1.INVALID_SORT_CLAUSE_PATH.toApiException(
-                "path ('%s') cannot start with '$' (except for pseudo-fields '$lexical', '$vector' and '$vectorize')",
-                path);
+      if (innerValue.isTextual()) {
+        throw ErrorCodeV1.INVALID_SORT_CLAUSE_VALUE.toApiException(
+            "Sort ordering value can be String only for Lexical or Vectorize search");
       }
+      throw ErrorCodeV1.INVALID_SORT_CLAUSE_VALUE.toApiException(
+          "Sort ordering value should be integer `1` or `-1`; or Array (Vector); or String (Lexical or Vectorize), was: %s",
+          JsonUtil.nodeTypeAsString(innerValue));
     }
-
-    try {
-      path = DocumentPath.verifyEncodedPath(path);
-    } catch (IllegalArgumentException e) {
-      throw ErrorCodeV1.INVALID_SORT_CLAUSE_PATH.toApiException(
-          "sort clause path ('%s') is not a valid path. " + e.getMessage(), path);
+    if (!(innerValue.intValue() == 1 || innerValue.intValue() == -1)) {
+      throw ErrorCodeV1.INVALID_SORT_CLAUSE_VALUE.toApiException(
+          "Sort ordering value can only be `1` for ascending or `-1` for descending (not `%s`)",
+          innerValue);
     }
 
-    return path;
+    boolean ascending = innerValue.intValue() == 1;
+    return SortExpression.sort(validatedPath, ascending);
```

Comment:
Just a suggestion - and in line with my confusion about the function name above. Is this the time to introduce a SortType enum and add it to the SortExpression ? Feels like we are doing a lot in the SortExpression what would be simplier with a type field 

Or subclassing to avoid chameleon class